text/metrics/0146-metrics-prototype-scenarios.md

```diff
@@ -0,0 +1,220 @@
+the store is running).
+
+The store has plenty supply of potato and tomato, with the following price:
+
+* Potato: $1.0 / ea
+* Tomato: $3.0 / ea
+
+Each customer has a unique name (e.g. customerA, customerB), a customer could
+come to the store multiple times. Here goes the Python snippet:
+
+```python
+store = GroceryStore("Portland")
+store.process_order("customerA", {"potato": 2, "tomato": 3})
+store.process_order("customerB", {"tomato": 10})
+store.process_order("customerC", {"potato": 2})
+store.process_order("customerA", {"tomato": 1})
+```
+
+When the store is closed, we will report the following metrics:
+
+### Order info
+
+| Store    | Customer  | Number of Orders | Amount (USD) |
+| -------- | --------- | ---------------- | ------------ |
+| Portland | customerA | 2                | 14.0         |
+| Portland | customerB | 1                | 30.0         |
+| Portland | customerC | 1                | 2.0          |
+
+### Items sold
+
+| Store    | Customer  | Item   | Count |
+| -------- | --------- | ------ | ----- |
+| Portland | customerA | potato | 2     |
+| Portland | customerA | tomato | 4     |
+| Portland | customerB | tomato | 10    |
+| Portland | customerC | potato | 2     |
+
+## Scenario 2: HTTP Server
+
+The _HTTP Server_ scenario covers how a library developer X could use metrics
+API to instrument a library, and how the application developer Y can configure
+the library to use OpenTelemetry SDK in a final application. X and Y are working
+for different companies and they don't communicate. The demo has two parts - the
+library (HTTP lib owned by X) and the server app (owned by Y):
+
+* How developer X could instrument the library code in a vendor agnostic way
+  * Performance is critical for X
+  * X doesn't know which metrics and which dimensions will Y pick
+  * X doesn't know the aggregation time window, nor the final destination of the
+    metrics
+* How developer Y could configure the SDK and exporter
+  * How should Y hook up the metrics SDK with the library
+  * How should Y configure the time window(s) and destination(s)
+  * How should Y pick the metrics and the dimensions
+
+### Library Requirements
+
+The library developer (developer X) will expose the following metrics out of
+box:
+
+### Pull Metrics
+
+These are pull metrics - the value is always available, and is only reported and
+collected based on the ask from consumer(s). If there is no ask from the
+consumer, the value will not be reported at all (e.g. there is no API call to
+fetch the room temperature unless someone is asking for the room temperature).
+
+#### Process CPU Usage
+
+Note: the **Host Name** should leverage [`OpenTelemetry
+Resource`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md),
+so it should be covered by the metrics SDK rather than API, and strictly
+speaking it is not considered as a "dimension" from the SDK perspective.
```

Comment:
I will ask about how vendors can "enrich" a data point.  I assume we have some "Processors" or extension points for vendors to enrich or alter data points (what is allowed/disallowed is also a discussion).

Separately, it would be good to clearly state who is responsible for providing the "Resource" labels?  Is it an Exporter or a Processor, or ???

Reply:
IIUC - This is done in Exporters for Trace.  If a vendor isn't directly supporting `Resouce` (or where their notion of `Resource` doesn't fully align with OTLP), the vendor can lift `Resource` labels onto the trace in the export method.  We actually plan to do this for instrumentation library (although I don't think we do it today).

Reply:
These are more like the SDK design question and implementation question. Probably not putting too much info in this doc since we're trying to cover the scenario/scope here?